forest/daemon/tests/keystore_tests.rs

```diff
@@ -42,18 +40,17 @@ fn forest_headless_no_encrypt_no_passphrase_should_succeed() -> Result<()> {
     Ok(())
 }
 
+/// We should prompt the user if they want to create a password
```

Comment:
I don't understand the reason behind the change in the logic here. Even if I provide the keystore phrase via an environmental variable, I should create my own phrase?

```
❯ FOREST_KEYSTORE_PHRASE=cthulhu target/release/forest --chain calibnet
2023-06-12T10:38:05.625877Z  INFO forest: Using default calibnet config
2023-06-12T10:38:05.626667Z  INFO forest::daemon: Starting Forest daemon, version 0.8.2+git.aaef338f0
2023-06-12T10:38:05.626688Z  INFO forest_libp2p::service: Could not decode networking keystore!
2023-06-12T10:38:05.626745Z  INFO forest_utils::io: Permissions set to 0600 on File { fd: 9, path: "/home/rumcajs/.local/share/forest/libp2p/keypair", read: false, write: true }
2023-06-12T10:38:05.626766Z  INFO forest::daemon: PeerId: 12D3KooWRsWRe4EGJ6B7UxZnXEJXcUP6p3v97dRAA1hHBmaFKbgs
2023-06-12T10:38:05.626772Z  WARN forest::daemon: Ignoring passphrase provided in FOREST_KEYSTORE_PHRASE for keystore creation
Create a password for Forest's keystore:
```

Reply:
I was surprised that we'd implicitly *create* (not load) a passphrase with no confirmation. It's relatively easy to lose the value of an environment variable (that's why they're often used for storing secrets), and I wouldn't want the user to end up with a system they didn't know the password to unless they had confirmed their password. I will back this change out if you think this is important.

Reply:
Unless something changed, this was also the only way to provide the passphrase in headless mode, i.e., setting up Forest and running it with some scripts. This is not really meant for a "front-end" user.

Reply:
Yep, added back :)